source/control_librariesConfig.cmake.in

```diff
@@ -1,34 +1,58 @@
 @PACKAGE_INIT@
 
-set(_control_libraries_supported_components state_representation dynamical_systems robot_model controllers)
-set(control_libraries_LIBRARIES "@INSTALL_INTERFACE_LINK_LIBRARIES@")
+set(_control_libraries_namespace "control_libraries::")
+set(_control_libraries_supported_components "@INSTALL_SUPPORTED_COMPONENTS@")
 
 include(CMakeFindDependencyMacro)
 find_dependency(Eigen3)
 
-# Find robot model dependencies if it is in the interface library list and no components are explicitly listed
-if ("robot_model" IN_LIST control_libraries_LIBRARIES AND NOT control_libraries_FIND_COMPONENTS)
-  find_dependency(pinocchio)
-  find_dependency(OsqpEigen)
-  find_dependency(osqp)
+set(_control_libraries_to_find ${control_libraries_FIND_COMPONENTS})
+if (NOT control_libraries_FIND_COMPONENTS)
+  set(_control_libraries_to_find ${_control_libraries_supported_components})
 endif()
 
-foreach(_comp ${control_libraries_FIND_COMPONENTS})
+set(control_libraries_LIBRARIES "")
+while(_control_libraries_to_find)
+  list(POP_FRONT _control_libraries_to_find _comp)
+
   if (${_comp} IN_LIST _control_libraries_supported_components)
     set(control_libraries_${_comp}_FOUND True)
   else()
     set_and_check(control_libraries_FOUND False)
     set_and_check(control_libraries_NOT_FOUND_MESSAGE "Unsupported component: ${_comp}")
   endif()
 
-  # Find robot model dependencies if the corresponding components are explicitly listed
-  if (${_comp} STREQUAL "controllers" OR ${_comp} STREQUAL "robot_model")
+  # Find robot model dependencies if the corresponding component is needed
+  # FIXME: this should be done automatically
+  if (${_comp} STREQUAL "robot_model")
     find_dependency(pinocchio)
```

Comment:
Do we need hpp-fcl here?

Reply:
Not sure. It built and ran correctly for me so I imagine `pinocchio` might be pulling it itself?